test/new-e2e/utils/infra/stack_manager.go

```diff
@@ -141,6 +142,17 @@ func (sm *StackManager) DeleteStack(ctx context.Context, name string) error {
 	return sm.deleteStack(ctx, name, stack)
 }
 
+func (sm *StackManager) ForceRemoveStack(ctx context.Context, name string) error {
```

Comment:
What is the reason to not use the existing function `deleteStack`?

Reply:
Delete stack would destroy the resources created by pulumi. This is not the intention here.
We destroy the resources manually in a [seperate job](https://github.com/DataDog/datadog-agent/blob/main/.gitlab/kernel_version_testing/system_probe.yml#L353).

The intention here is to force remove the stack, so that the cleaner job which runs periodically does not consistently fail on leftover stale stacks from `kernel_matrix_testing` jobs.

For further contexts. we do not use the delete stack at all for cleanup, because we want to avoid any chance of having leftover instances due to a failure in the delete operation. Instead we manually destroy the ec2 instances as you can see in the cleanup jobs.